### ISSUE TITLE
A couple of errors with PyQt6

### ISSUE THREAD

**Issue by fxjaeckel** [spyder-ide/spyder#22830]
### What steps will reproduce the problem?

Install PyQt6 and PyQt6-WebEngine using pip.
Start spyder.
Execute piece of code using IPython console (F5).
Code executes, but Issue reporter says: "Spyder has encountered an internal problem!"

### Traceback
```python-traceback
TypeError: invalid result from _SpyderButtonsProxyStyle.styleHint(), 'ButtonLayout' object cannot be interpreted as an integer
TypeError: invalid result from _SpyderButtonsProxyStyle.styleHint(), 'ButtonLayout' object cannot be interpreted as an integer
TypeError: invalid result from _SpyderButtonsProxyStyle.styleHint(), 'ButtonLayout' object cannot be interpreted as an integer
TypeError: invalid result from _SpyderButtonsProxyStyle.styleHint(), 'ButtonLayout' object cannot be interpreted as an integer
Traceback (most recent call last):
  File "/home/jaeckel/.pyenv/versions/3.11.9/lib/python3.11/site-packages/spyder/plugins/editor/widgets/base.py", line 1114, in mousePressEvent
    event = QMouseEvent(QEvent.MouseButtonPress, event.pos(),
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: arguments did not match any overloaded call:
  QMouseEvent(type: QEvent.Type, localPos: QPointF, button: Qt.MouseButton, buttons: Qt.MouseButton, modifiers: Qt.KeyboardModifier, device: Optional[QPointingDevice] = QPointingDevice.primaryPointingDevice()): argument 2 has unexpected type 'QPoint'
  QMouseEvent(type: QEvent.Type, localPos: QPointF, globalPos: QPointF, button: Qt.MouseButton, buttons: Qt.MouseButton, modifiers: Qt.KeyboardModifier, device: Optional[QPointingDevice] = QPointingDevice.primaryPointingDevice()): argument 2 has unexpected type 'QPoint'
  QMouseEvent(type: QEvent.Type, localPos: QPointF, scenePos: QPointF, globalPos: QPointF, button: Qt.MouseButton, buttons: Qt.MouseButton, modifiers: Qt.KeyboardModifier, device: Optional[QPointingDevice] = QPointingDevice.primaryPointingDevice()): argument 2 has unexpected type 'QPoint'
Traceback (most recent call last):
  File "/home/jaeckel/.pyenv/versions/3.11.9/lib/python3.11/site-packages/spyder/plugins/editor/widgets/base.py", line 1114, in mousePressEvent
    event = QMouseEvent(QEvent.MouseButtonPress, event.pos(),
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: arguments did not match any overloaded call:
  QMouseEvent(type: QEvent.Type, localPos: QPointF, button: Qt.MouseButton, buttons: Qt.MouseButton, modifiers: Qt.KeyboardModifier, device: Optional[QPointingDevice] = QPointingDevice.primaryPointingDevice()): argument 2 has unexpected type 'QPoint'
  QMouseEvent(type: QEvent.Type, localPos: QPointF, globalPos: QPointF, button: Qt.MouseButton, buttons: Qt.MouseButton, modifiers: Qt.KeyboardModifier, device: Optional[QPointingDevice] = QPointingDevice.primaryPointingDevice()): argument 2 has unexpected type 'QPoint'
  QMouseEvent(type: QEvent.Type, localPos: QPointF, scenePos: QPointF, globalPos: QPointF, button: Qt.MouseButton, buttons: Qt.MouseButton, modifiers: Qt.KeyboardModifier, device: Optional[QPointingDevice] = QPointingDevice.primaryPointingDevice()): argument 2 has unexpected type 'QPoint'
```

## Versions

* Spyder version: 6.0.2  (pip)
* Python version: 3.11.9 64-bit
* Qt version: 6.7.1
* PyQt6 version: 6.7.1
* Operating System: Linux-6.8.0-48-generic-x86_64-with-glibc2.39

### Dependencies

```
# Mandatory:
aiohttp >=3.9.3                  :  3.10.5 (OK)
asyncssh >=2.14.0,<3.0.0         :  2.17.0 (OK)
atomicwrites >=1.2.0             :  1.4.1 (OK)
chardet >=2.0.0                  :  5.2.0 (OK)
cloudpickle >=0.5.0              :  3.0.0 (OK)
cookiecutter >=1.6.0             :  2.6.0 (OK)
diff_match_patch >=20181111      :  20230430 (OK)
github >=2.3.0                   :  2.4.0 (OK)
importlib_metadata >=4.6.0       :  8.3.0 (OK)
intervaltree >=3.0.2             :  3.1.0 (OK)
IPython >=8.13.0,<9.0.0,!=8.17.1 :  8.26.0 (OK)
jedi >=0.17.2,<0.20.0            :  0.19.1 (OK)
jellyfish >=0.7                  :  1.1.0 (OK)
jsonschema >=3.2.0               :  4.23.0 (OK)
keyring >=17.0.0                 :  25.3.0 (OK)
nbconvert >=4.0                  :  7.16.4 (OK)
numpydoc >=0.6.0                 :  1.8.0 (OK)
parso >=0.7.0,<0.9.0             :  0.8.4 (OK)
pexpect >=4.4.0                  :  4.9.0 (OK)
pickleshare >=0.4                :  0.7.5 (OK)
psutil >=5.3                     :  6.0.0 (OK)
pygments >=2.0                   :  2.18.0 (OK)
pylint >=3.1,<4                  :  3.2.6 (OK)
pylint_venv >=3.0.2              :  3.0.3 (OK)
pyls_spyder >=0.4.0              :  0.4.0 (OK)
pylsp >=1.12.0,<1.13.0           :  1.12.0 (OK)
pylsp_black >=2.0.0,<3.0.0       :  2.0.0 (OK)
pyuca >=1.2                      :  1.2 (OK)
qdarkstyle >=3.2.0,<3.3.0        :  3.2.3 (OK)
qstylizer >=0.2.2                :  0.2.3 (OK)
qtawesome >=1.3.1,<1.4.0         :  1.3.1 (OK)
qtconsole >=5.6.1,<5.7.0         :  5.6.1 (OK)
qtpy >=2.4.0                     :  2.4.1 (OK)
rtree >=0.9.7                    :  1.3.0 (OK)
setuptools >=49.6.0              :  73.0.0 (OK)
sphinx >=0.6.6                   :  8.0.2 (OK)
spyder_kernels >=3.0.1,<3.1.0    :  3.0.1 (OK)
superqt >=0.6.2,<1.0.0           :  0.6.7 (OK)
textdistance >=4.2.0             :  4.6.2 (OK)
three_merge >=0.1.1              :  0.1.1 (OK)
watchdog >=0.10.3                :  4.0.2 (OK)
xdg >=0.26                       :  0.28 (OK)
yarl >=1.9.4                     :  1.11.1 (OK)
zmq >=24.0.0                     :  26.1.1 (OK)

# Optional:
cython >=0.21                    :  3.0.11 (OK)
matplotlib >=3.0.0               :  3.9.2 (OK)
numpy >=1.7                      :  2.0.1 (OK)
pandas >=1.1.1                   :  2.2.2 (OK)
scipy >=0.17.0                   :  1.14.0 (OK)
sympy >=0.7.3                    :  1.13.2 (OK)
```

**Comment by ccordoba12** [comment]
Hey @fxjaeckel, thanks for reporting. I was able to easily reproduce your problems with PyQt6.

We'll try to fix them in our next release (6.0.3), to be released at the end of the month.

**Comment by ccordoba12** [comment]
Sorry, this will have to wait until 6.1.0 because we don't officially support Qt6 and it's not critical.

**Comment by fxjaeckel** [comment]
The  "TypeError: invalid result" seems like its fixed in PR #22620
The "TypeError: arguments did not match any overloaded call" should be fixed with PR #22846
